daemon/logger/local/read.go

```diff
@@ -96,51 +96,81 @@ func getTailReader(ctx context.Context, r loggerutils.SizeReaderAt, req int) (io
 	return io.NewSectionReader(r, offset, size), found, nil
 }
 
-func decodeFunc(rdr io.Reader) func() (*logger.Message, error) {
-	proto := &logdriver.LogEntry{}
-	buf := make([]byte, initialBufSize)
```

Comment:
It should probably be `make([]byte, 0, initialBufSize)` unless you want buf to be filled with `\0`s

Reply:
I do want it to be filled with 0's since the reader interface needs it.

Reply:
I assume same is true for other similar places (i.e. you always want buffer to be filled with `\0`s)

Reply:
Yes, anywhere we are creating a buffer to be used with a reader.